code/datums/recipe.dm

```diff
@@ -133,6 +133,11 @@
 				O.reagents.del_reagent(r_r)
 			//Transfer any reagents found in the object, to the resulting object
 			O.reagents.trans_to(result_obj, O.reagents.total_volume)
+		//Transfer any luckiness from the ingredients, to the resulting object
+		var/obj/item/I = O
+		var/obj/item/result_item = result_obj
+		if(I.luckiness)
+			result_item.luckiness += I.luckiness
```

Comment:
this will runtime when you cook a non-item food like the pop out cake

Reply:
Added a check for both the ingredient and the dish.